salt/modules/win_pkg.py

```diff
@@ -1188,6 +1188,9 @@ def install(name=None, refresh=False, pkgs=None, **kwargs):
                 'extra_install_flags': kwargs.get('extra_install_flags')
             }
         }
+    elif len(pkg_params) == 1:
+        for pkg in pkg_params:
```

Comment:
why the for, if you know it has exactly one item?

Reply:
d = {'cat' : {'name': 'bob’'}}
# other ways of doing it
1. d[list(d)[0]]['id'] = 404
2. d[d.keys()[0]][‘id’] = 404
3. next(d.iteritems())[1]['id'] = 404

I believe the for loop to be more graceful and effective. Ways 1 and 2 both make a new list rather then using a generator that uses an item that already exist. Which I understand to be slower. It also makes more grabbage. 3 could be an option. Personally I like the for loop over it because of this “import this”. Please let me know if you know a better way... I'm always willing to learn.

Reply:
pkg_params[pkgs[0]]['extra_install_flags'] = kwargs.get('extra_install_flags')
this works